src/evalfunc.c

```diff
@@ -5216,6 +5211,47 @@ f_get(typval_T *argvars, typval_T *rettv)
 	copy_tv(tv, rettv);
 }
 
+/*
+ * "getcellpixels()" function
+ */
+    static void
+f_getcellpixels(typval_T *argvars UNUSED, typval_T *rettv)
+{
+    if (rettv_list_alloc(rettv) == FAIL)
+        return;
+
+#if defined(FEAT_GUI)
+    if (gui.in_use)
+    {
+        // success pixel size and no gui.
+        list_append_number(rettv->vval.v_list, (varnumber_T)gui.char_width);
+        list_append_number(rettv->vval.v_list, (varnumber_T)gui.char_height);
+    }
+    else
+    {
+#endif
+        struct cellsize cs;
+# if defined(UNIX)
+        mch_calc_cell_size(&cs);
+# else
+        // Non-Unix CUIs are not supported, so set this to -1x-1.
+        cs.cs_xpixel = -1;
+        cs.cs_ypixel = -1;
+# endif
+
+        // failed get pixel size.
+        if (cs.cs_xpixel == -1)
+            return;
+
+        // success pixel size and no gui.
+        list_append_number(rettv->vval.v_list, (varnumber_T)cs.cs_xpixel);
+        list_append_number(rettv->vval.v_list, (varnumber_T)cs.cs_ypixel);
+#if defined(FEAT_GUI)
+    }
+#endif
```

Comment:
Your code is currently:
```C
#if defined(FEAT_GUI)
    if (...)
    {
        ...
    }
    else
    {
#endif
        ...
#if defined(FEAT_GUI)
    }
#endif
```
But I think that the following style is better:
```C
#if defined(FEAT_GUI)
    if (...)
    {
        ...
    }
    else
#endif
    {
        ...
    }
```
This can reduce the ifdefs.

Reply:
Fixed in 7818a497285a0c416703698c4c20cc3006acbe75.